engine/src/cubos/engine/tools/palette_editor/plugin.cpp

```diff
@@ -0,0 +1,79 @@
+#include <cubos/core/data/old/debug_serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/renderer/plugin.hpp>
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+#include <cubos/engine/tools/palette_editor/plugin.hpp>
+#include <cubos/engine/voxels/palette.hpp>
+#include <cubos/engine/voxels/plugin.hpp>
+
+#include "imgui.h"
+
+using cubos::core::data::old::Debug;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using namespace cubos::engine;
+
+struct SelectedPaletteInfo
+{
+    VoxelPalette palette;
+};
+
+static void checkAssetEventSystem(cubos::core::ecs::EventReader<tools::AssetSelectedEvent> reader, Read<Assets> assets,
+                                  Write<SelectedPaletteInfo> selectedPalette)
+{
+    for (const auto& event : reader)
+    {
+        if (assets->type(event.asset) == typeid(VoxelPalette))
+        {
+            CUBOS_INFO("Opening Palette Editor for {} ...", Debug(event.asset));
+            const Asset<VoxelPalette> paletteAsset = AnyAsset(event.asset);
+            selectedPalette->palette = assets->read(paletteAsset).get();
+        }
+    }
+}
+
+static void paletteEditor(Write<Renderer> renderer, Write<SelectedPaletteInfo> selectedPalette)
+{
+    auto& palette = selectedPalette->palette;
+
+    if (palette.data() == nullptr)
+    {
+        return; // no data on the palette yet!
+    }
+
+    ImGui::Begin("Palette Editor");
```

Comment:
There should be something identifying the palette at the top, along with whether or not that palette is the one currently active.